rio/frontend/parser.mly

```diff
@@ -67,16 +65,20 @@
 
 %%
 
+set:
+    | CLSS                                              { Class($1) }
+    | UNION LBRACKET; pl = setlist; RBRACKET            { Union pl }
+
 /* Policies */
 policy:
-    | FIFO LBRACKET; pl = arglist; RBRACKET             { Fifo pl }
-    | STRICT LBRACKET; pl = arglist; RBRACKET           { Strict pl }
-    | RR LBRACKET; pl = arglist; RBRACKET               { RoundRobin pl }
-    | WFQ LBRACKET; pl = weighted_arglist; RBRACKET     { WeightedFair pl }
-    | EDF LBRACKET; pl = arglist; RBRACKET              { EarliestDeadline pl }
-    | SJN LBRACKET; pl = arglist; RBRACKET              { ShortestJobNext pl }
-    | SRTF LBRACKET; pl = arglist; RBRACKET             { ShortestRemaining pl }
-    | RCSP LBRACKET; pl = arglist; RBRACKET             { RateControlled pl }
+    | FIFO LBRACKET; pl = set; RBRACKET                                              { Fifo pl }
+    | EDF LBRACKET; pl = set; RBRACKET                                               { EarliestDeadline pl }
+    | SJN LBRACKET; pl = set; RBRACKET                                               { ShortestJobNext pl }
+    | SRTF LBRACKET; pl = set; RBRACKET                                              { ShortestRemaining pl }
+    | RR LBRACKET; pl = arglist; RBRACKET                                            { RoundRobin pl }
+    | STRICT LBRACKET; pl = arglist; RBRACKET                                        { Strict pl }
+    | WFQ LBRACKET; pl = arglist; RBRACKET; LBRACKET; wt = weight_list; RBRACKET     { WeightedFair (pl, wt) }
```

Comment:
What do you think about changing this so the syntax for WFQ still looks like 
```
WFQ[(stream_A, weight_A), (stream_B, weight_B), (stream_C, weight_C)]
```
instead of 
```
WFQ[stream_A, stream_B, stream_C][weight_A, weight_B, weight_C]
```
Even if we keep the `WFQ` constructor in `Ast` as `WFQ of stream list * int list`, it should be simple to keep the user syntax the old way. The advantage with this is that our parsing still guarantees our two lists have equal length.

(I also subjectively think the old way looks nicer, but that's neither here nor there)

Reply:
Sure, happy to revert it to the old way! I think we can rely on the lists staying the same length for actual compilation.